src/tuesday/core/summaries/powerspectra.py

```diff
@@ -14,45 +18,43 @@
 
 
 def calculate_ps(  # noqa: C901
-    lc,
-    lc_redshifts,
-    box_length,
-    box_side_shape=None,
-    zs=None,
-    chunk_size=None,
-    chunk_skip=37,
-    calc_2d=True,
-    nbins=50,
-    k_weights=ignore_zero_ki,
-    postprocess=True,
-    kpar_bins=None,
-    log_bins=True,
-    crop=None,
-    calc_1d=False,
-    nbins_1d=14,
-    calc_global=False,
-    mu=None,
-    bin_ave=True,
-    interp=None,
-    prefactor_fnc=power2delta,
-    interp_points_generator=None,
-    get_variance=False,
-):
+    lc: un.Quantity,
+    lc_redshifts: np.ndarray,
+    box_length: un.Quantity,
+    zs: float | np.ndarray | None = None,
+    chunk_size: int | None = None,
+    chunk_skip: int | None = 37,
+    calc_2d: bool | None = True,
+    nbins: int | None = None,
+    k_weights: Callable | None = ignore_zero_ki,
+    postprocess: bool | None = True,
+    kpar_bins: int | np.ndarray | None = None,
+    log_bins: bool | None = True,
+    crop: list | np.ndarray | None = None,
+    calc_1d: bool | None = False,
+    nbins_1d: int | None = None,
+    calc_global: bool | None = False,
+    mu: float | None = None,
+    bin_ave: bool | None = True,
+    interp: bool | None = None,
+    prefactor_fnc: Callable | None = power2delta,
+    interp_points_generator: Callable | None = None,
+    get_variance: bool | None = False,
+) -> dict:
     r"""Calculate power spectra from a lightcone.
 
     Parameters
     ----------
-    lc : np.ndarray
-        The lightcone whose power spectrum we want to calculate.
-        The lightcone should be a 3D array with shape
-        [box_side_shape, box_side_shape, len(lc_redshifts)].
+    lc : un.Quantity
+        The 3D box whose power spectrum we want to calculate.
+        The box 3D array with shape
+        [X, Y, len(lc_redshifts)].
     lc_redshifts : np.ndarray
-        The redshifts of the lightcone.
-    box_length : float
-        The side length of the box in cMpc.
-    box_side_shape : int, optional
-        The number of pixels in one side of the box
-        (HII_DIM parameter in 21cmFAST).
+        The values at the last axis of the `lc` array,
+        typically the redshifts for a lightcone.
+    box_length : un.Quantity
+        The side length of the box.
+        Accepted units are: Mpc and Mpc/h.
     zs : np.ndarray, optional
```

Comment:
I think it would be way easier to instead choose the *indices* that you want to use in the PS, not the `zs`